cis194/week1/tien/homework1/src/Lib.hs

```diff
@@ -0,0 +1,30 @@
+module Lib
+  ( toDigits
+  , toDigitsRev
+  , buildWeights
+  , doubleEveryOther
+  , sumDigits
+  , validate
+  ) where
+
+-- Exercise One: --
+toDigits :: Integer -> [Integer]
+toDigits num
+  | num > 0 = map (read . (: [])) $ show num :: [Integer]
+  | otherwise = []
+
+toDigitsRev :: Integer -> [Integer]
+toDigitsRev num
+  | num > 0 = reverse $ toDigits num
+  | otherwise = []
+
+-- Exercise Two: --
+buildWeights :: Integer -> [Integer]
+buildWeights num = reverse $ map ((+ 1) . (`mod` 2)) [0 .. num - 1]
```

Comment:
We can have a look at `cycle` function, I think it'll greatly reduce the code here.
In addition to above, since `cycle` will give infinite list, please think about when to use `reverse`, should we use it here or on the list we are doubling against?